clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -1932,7 +1932,7 @@ void CodeGenFunction::GenerateCode(GlobalDecl GD, llvm::Function *Fn,
     // Implicit copy-assignment gets the same special treatment as implicit
     // copy-constructors.
     emitImplicitAssignmentOperatorBody(Args);
-  } else if (FD->hasAttr<OpenCLKernelAttr>() &&
+  } else if (getLangOpts().OpenCL && FD->hasAttr<OpenCLKernelAttr>() &&
```

Comment:
Why do we handle `OpenCLKernelAttr` in SYCL language mode? We have SYCL kernel attribute for SYCL kernels.
I would expect Sema to reject uses of `OpenCLKernelAttr` attribute for non-OpenCL languages, so adding `getLangOpts().OpenCL` should be unnecessary.

Reply:
The usptream uses new attribute sycl_entry_point and generates kernels based on its presence. I'm not sure OpenCLKernelAttr is being added somewhere in that approach. What we need to do is gradually switch from sycl_kernel to sycl_entry_point attribute once there are enough functionality supported, including handling sycl_entry_point as a `DeviceKernelAttr`.